### DISCUSSION TITLE
Update gasLimit after launching network.

### DISCUSSION THREAD

**Discussion by devsMynting** [0xPolygon/polygon-edge#634] [Q&A]
Is it possible to update the gasLimit of a polygon-edge chain that has already been launched? If so, how could it be achieved?

**Comment by laviniat1996** [comment]
Hi @devsMynting ,
Thank you for your question!

You can use the `--price-limit` flag to set the gas limit for each node (the value can be different for each node), even if the chain has already been launched.  An example on how to use it would be: `polygon-edge server --price-limit 100000`

You can find more about it [here](https://docs.polygon.technology/docs/edge/get-started/set-up-ibft-on-the-cloud) (scroll to the end of the page) and [here](https://docs.polygon.technology/docs/edge/get-started/cli-commands#server-flags) .

**Reply by devsMynting** [reply]
Thanks for the response. Though I understand the price-limit for gas can be set on a per node basis; however, does this also update the gasLimit of each block as specified in --block-gas-limit in genesis.json (the default value is set to 0x500000)?

**Comment by Kourin1996** [comment]
@devsMynting Hello, block gas limit is adjusted every block production. Each node has block gas target setting which is given from `--block-gas-target` of `server` command. If parent block's gas limit is different from node's block gas target, block producer can update block gas limit. (But it can add/subtract only (parent block gas limit)/1024 from parent block gas limit per block). If block gas target is zero, then the node sets parent block gas limit to next block gas limit.

Ideally, all nodes has same block gas target value, block gas limit converges to the value. Or block gas limit won't change if all nodes' block gas target is zero.

In addition,
`--block-gas-limit` is set the block gas limit on genesis block, it means block gas limit begins from this value.

`--price-limit` is the threshold of transaction gas price to accept transaction for node.
